### PR TITLE
 block: Block::is_basic_valid 

### PR DESCRIPTION
**Pull request by sorpaas**
Most of the block parameters require either parent block or SputnikVM to check validity. So `is_basic_valid` only checks things that don't require them, that is, the transactions_root and the ommers_hash.

### CODE REVIEW

File: block/src/block.rs [hunk 1]
```diff
@@ -35,6 +35,20 @@ pub struct Block {
     pub ommers: Vec<Header>,
 }
 
+impl Block {
+    pub fn is_basic_valid(&self) -> bool {
+        if transactions_root(&self.transactions) != self.header.transactions_root {
```
Comment (whilei) [inline block/src/block.rs:40]: Is it ever possible that `self.header` is nil?

**Review by whilei — APPROVED** [review]

File: block/src/block.rs (hunk 1 above)
Comment (sorpaas) [reply block/src/block.rs:40]: Nope. Guaranteed by Rust's type system~~

Comment (sorpaas) [reply block/src/block.rs:40]: To get nil one must use the Option type explicitly. And if ever happen that one forgot to handle nil, the compiler would refuse the comply. :)

Comment (whilei) [reply block/src/block.rs:40]: That's pretty awesome.

**Pull request merged**